lib/rubygems/command.rb

```diff
@@ -456,7 +456,9 @@ def add_extra_args(args)
     until extra.empty? do
       ex = []
       ex << extra.shift
-      ex << extra.shift if extra.first.to_s =~ /^[^-]/
+      if (!extra.first.to_s.empty? && !extra.first.to_s.start_with?("-"))
```

Comment:
Do you feel this code is better than the previous code? I think the previous code was easier to grasp, so perhaps we should ignore the cop here? Or at least invert the condition so it reads a bit simpler? (`!A && !B = A || B`).

Reply:
(`nil` and `""` must not enter the conditional.) I reverted, added a disable.